dotcom-rendering/src/components/Discussion.tsx

```diff
@@ -85,6 +85,96 @@ const remapFilters = (
 	} satisfies FilterOptions;
 };
 
+type State = {
+	comments: CommentType[];
+	isClosedForComments: boolean;
+	isExpanded: boolean;
+	commentPage: number;
+	filters: FilterOptions;
+	hashCommentId: number | undefined;
+	totalPages: number;
+	loading: boolean;
+};
+
+const initialState: State = {
+	comments: [],
+	isClosedForComments: false,
+	isExpanded: false,
+	commentPage: 1,
+	filters: initFiltersFromLocalStorage(),
+	hashCommentId: undefined,
+	totalPages: 0,
+	loading: true,
+};
+
+type Action =
+	| {
+			type: 'commentsLoaded';
+			comments: CommentType[];
+			isClosedForComments: boolean;
+			totalPages: number;
+	  }
+	| { type: 'expandComments' }
+	| { type: 'addComment'; comment: CommentType }
+	| { type: 'updateCommentPage'; commentPage: number; shouldExpand: boolean }
+	| { type: 'updateHashCommentId'; hashCommentId: number | undefined }
+	| { type: 'filterChange'; filters: FilterOptions; commentPage?: number }
+	| { type: 'updateTotalPages'; totalPages: number }
+	| { type: 'setLoading'; loading: boolean };
+
+const reducer = (state: State, action: Action): State => {
+	switch (action.type) {
+		case 'commentsLoaded':
+			return {
+				...state,
+				comments: action.comments,
+				isClosedForComments: action.isClosedForComments,
+				totalPages: action.totalPages,
+				loading: false,
+			};
+		case 'addComment':
+			return {
+				...state,
+				comments: [action.comment, ...state.comments.slice(0, -1)], // Remove last item from our local array Replace it with this new comment at the start
```

Comment:
Yep, I think you're right, perhaps it would be better to simply add it to the array (and show 26/51/101 comments) to the user until they refreshed or navigated to a different page so we don't lose any content. It would be the count is incorrect though. 

To stop this PR growing to much, I think it would be better to address this in a separate PR so I'll ticket this up.

Reply:
This is fixed here https://github.com/guardian/dotcom-rendering/pull/10389